tests/test_middleware_comprehensive.py

```
@@ -0,0 +1,436 @@
+        # Get the same tracker
+        tracker2 = middleware.get_tracker("op-1")
+        assert tracker1 is tracker2
+
+        # Try to get non-existent tracker
+        tracker3 = middleware.get_tracker("non-existent")
+        assert tracker3 is None
+
+    def test_progress_middleware_remove_tracker(self, middleware):
+        """Test removing trackers."""
+        # Create a tracker
+        tracker = middleware.create_tracker("removable", "Removable Op")
```

Ruff (0.12.2)

Local variable `tracker` is assigned to but never used

Remove assignment to unused variable `tracker`

(F841)